metrics_sdk/lib/opentelemetry/sdk/metrics/aggregation/sum.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Aggregation
+        # Contains the implementation of the Sum aggregation
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#sum-aggregation
+        class Sum
+          def initialize
+            @aggregation_temporality = :delta
+
+            @data_points = {}
+            @mutex = Mutex.new
+          end
+
+          def collect(start_time, end_time)
+            ndps = @data_points.map do |_key, ndp|
+              ndp.start_time_unix_nano = start_time
+              ndp.time_unix_nano = end_time
+              ndp
+            end
+            @data_points.clear if @aggregation_temporality == :delta
+            ndps
+          end
+
+          def update(increment, attributes)
+            ndp = @data_points[attributes] || @data_points[attributes] = NumberDataPoint.new(
+              attributes,
+              nil,
+              nil,
+              0,
+              nil,
+              nil
+            )
```

Comment:
How do we feel about keyword init for structs?

Reply:
🤷 we didn't use it for `SpanData`: https://github.com/open-telemetry/opentelemetry-ruby/blob/6fed21de005917f1900a8b8dc879f89623e4c61a/sdk/lib/opentelemetry/sdk/trace/span.rb#L255-L276

I do like keyword arguments in general. I'm ambivalent about them for internal things like this that are called only in one place. I would expect our tests to quickly catch a bug here. 

If folks have strong feelings about this, we should change `to_span_data` as well, for consistency.

Reply:
I don't have super strong feelings, other than "it's hard to remember what the positional args mean". I don't care enough to block over it though, just thought I'd take the temperature of the room. :)

Reply:
For the most part, we've had 2 callers (the `to_span_data` method and a test helper, although the latter has been copied a bunch of times). I think that's true for metrics as well. With only 2 callers, positional args don't seem that onerous. I'm not averse to keyword args, though, unless there's runtime overhead associated with them.